File: bench/io/bench_write.nim
# Write a million formatted lines. See bench_write.c for what this measures.
import std/[times, os]

const LINES = 1000000
let t0 = epochTime()
var f: File
if not open(f, "io_out_nim.txt", fmWrite):
  quit(1)
for i in 0 ..< LINES:
  f.write($i & " payload " & $(i * 7 mod 9973) & "\n")
close(f)
let ms = int((epochTime() - t0) * 1000)

let size = getFileSize("io_out_nim.txt")
removeFile("io_out_nim.txt")
echo "result: ", size
echo "elapsed: ", ms

File: bench/io/bench_write.lua
-- Write a million formatted lines. See bench_write.c for what this measures.
local LINES = 1000000
local t0 = os.clock()
local f = io.open("io_out_lua.txt", "wb")
for i = 0, LINES - 1 do
    f:write(i, " payload ", (i * 7) % 9973, "\n")
end
f:close()
local ms = (os.clock() - t0) * 1000
local r = io.open("io_out_lua.txt", "rb")
local size = r:seek("end")
r:close()
os.remove("io_out_lua.txt")
print(string.format("result: %d", size))
print(string.format("elapsed: %.0f", ms))
